cmake/bitcoin-config.h.in

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_CONFIG_H
+#define BITCOIN_CONFIG_H
+
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "@PACKAGE_NAME@"
+
+/* Define to the version of this package. */
+#define PACKAGE_VERSION "@PACKAGE_VERSION@"
+
+/* Major version. */
+#define CLIENT_VERSION_MAJOR @CLIENT_VERSION_MAJOR@
+
+/* Minor version. */
+#define CLIENT_VERSION_MINOR @CLIENT_VERSION_MINOR@
+
+/* Version build. */
+#define CLIENT_VERSION_BUILD @CLIENT_VERSION_BUILD@
```

Comment:
Any reason to change the order and at the same the time the docstrings? This makes it harder to review the diff:


```diff
diff --git a/tmp/a b/tmp/b
index 13d6b32ec..bd846a352 100644
--- a/tmp/a
+++ b/tmp/b
@@ -1,442 +1,44 @@
-/* src/config/bitcoin-config.h.  Generated from bitcoin-config.h.in by configure.  */
-/* src/config/bitcoin-config.h.in.  Generated from configure.ac by autoheader.  */
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 #ifndef BITCOIN_CONFIG_H
-
 #define BITCOIN_CONFIG_H
 
-/* Define if building universal (internal helper macro) */
-/* #undef AC_APPLE_UNIVERSAL_BUILD */
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "Bitcoin Core"
 
-/* Define this symbol if type char equals int8_t */
-/* #undef CHAR_EQUALS_INT8 */
+/* Define to the version of this package. */
+#define PACKAGE_VERSION "24.99.0"
 
-/* Version Build */
-#define CLIENT_VERSION_BUILD 0
-
-/* Version is release */
-#define CLIENT_VERSION_IS_RELEASE false
-
-/* Major version */
+/* Major version. */
 #define CLIENT_VERSION_MAJOR 24
 
-/* Minor version */
+/* Minor version. */
 #define CLIENT_VERSION_MINOR 99
 
-/* Copyright holder(s) before %s replacement */
-#define COPYRIGHT_HOLDERS "The %s developers"
+/* Version build. */
+#define CLIENT_VERSION_BUILD 0
 
-/* Copyright holder(s) */
-#define COPYRIGHT_HOLDERS_FINAL "The Bitcoin Core developers"
-
-/* Replacement for %s in copyright holders string */
-#define COPYRIGHT_HOLDERS_SUBSTITUTION "Bitcoin Core"
-
-/* Copyright year */
-#define COPYRIGHT_YEAR 2023
-
-/* Define this symbol to build code that uses ARMv8 SHA-NI intrinsics */
-/* #undef ENABLE_ARM_SHANI */
-
...

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/27060#issuecomment-1425893451).